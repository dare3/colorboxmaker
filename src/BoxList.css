/* BoxList.css */
.box-list {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
  
  .boxes-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  h2 {
    margin-bottom: 20px;
  }
  