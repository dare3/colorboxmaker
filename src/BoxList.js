// BoxList.js
import './BoxList.css';
import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import { v4 as uuid } from 'uuid';

function BoxList() {
  const [boxes, setBoxes] = useState([]);

  const addBox = (newBox) => {
    setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }]);
  };

  const removeBox = (id) => {
    setBoxes(boxes => boxes.filter(box => box.id !== id));
  };

  return (
    <div className="box-list">
      <h2>Create a New Box</h2>
      <NewBoxForm addBox={addBox} />
      <div className="boxes-container">
        {boxes.map(box => (
          <Box 
            key={box.id} 
            id={box.id} 
            width={box.width} 
            height={box.height} 
            backgroundColor={box.backgroundColor} 
            removeBox={removeBox} 
          />
        ))}
      </div>
    </div>
  );
}

export default BoxList;
