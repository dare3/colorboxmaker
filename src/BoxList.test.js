import React from 'react';
import { render } from '@testing-library/react';
import BoxList from './BoxList';

test('renders BoxList component', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
