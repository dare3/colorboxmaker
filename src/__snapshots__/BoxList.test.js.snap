// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoxList component 1`] = `
<DocumentFragment>
  <div
    class="box-list"
  >
    <h2>
      Create a New Box
    </h2>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="boxes-container"
    />
  </div>
</DocumentFragment>
`;
